### PR TITLE
Fix / 9156 Gate by Feature Flag

### PR DESCRIPTION
**Pull request by 10upsimon**
## Summary

Addresses issue:

- #9156

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.
- [ ] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by zutigrm — APPROVED** [review]
Thanks @10upsimon LGTM

File: assets/js/components/KeyMetrics/key-metrics-widgets.js [hunk 1]
```diff
@@ -412,6 +412,7 @@ const KEY_METRICS_WIDGETS = {
 			'google-site-kit'
 		),
 		requiredConversionEventName: [ 'add_to_cart' ],
+		displayInList: shouldDisplayWidgetWithConversionEvent,
```
Comment (eugene-manuilov) [inline assets/js/components/KeyMetrics/key-metrics-widgets.js:415]: @10upsimon, I suppose the idea here is to ensure that this widget doesn't appear when the `conversionReporting` feature flag is not enabled for the site, right? The problem is that the `shouldDisplayWidgetWithCustomDimensions` function doesn't check for that feature flag, hence this widget will continue appearing. Or I miss something?

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]

File: assets/js/components/KeyMetrics/key-metrics-widgets.js (hunk 1 above)
Comment (10upsimon) [reply assets/js/components/KeyMetrics/key-metrics-widgets.js:415]: @eugene-manuilov the very first thing the function checks for is the feature flag?

```
	if ( ! isFeatureEnabled( 'conversionReporting' ) ) {
		return false;
	}
```

Comment (zutigrm) [reply assets/js/components/KeyMetrics/key-metrics-widgets.js:415]: @eugene-manuilov That function checks for feture flag before checking for detected events https://github.com/google/site-kit-wp/blob/0525c3b6f74dd88d924de35ef7658ad67ce4e774/assets/js/components/KeyMetrics/shouldDisplayWidgetWithConversionEvent.js#L25-L27

Comment (eugene-manuilov) [reply assets/js/components/KeyMetrics/key-metrics-widgets.js:415]: Yeah, you are right guys. I looked at the wrong function. Looks good now.

**Review by eugene-manuilov — APPROVED** [review]

**Pull request merged**